tools/clang/include/clang/Basic/DiagnosticSemaKinds.td

```diff
@@ -7471,6 +7471,8 @@ def err_hlsl_attribute_expects_string_literal_from_list: Error<
   "attribute %0 must have one of these values: %1">;
 def err_hlsl_attribute_valid_on_function_only: Error<
   "attribute is valid only on functions">;
+def err_hlsl_attribute_mismatch: Error<
+  "Invalid shader stage attribute combination">;
```

Comment:
nit: diagnostics should start with lowercase letters

Also, I wonder if there isn't a better way to say this. The existing `warn_duplicate_attribute` warning is similar. We also might want to add a `note` pointing at previous attributes.